internal/database/repository/queue.go

```diff
@@ -82,7 +88,18 @@ func (s *PostgresRepository) UpsertQueues(ctx context.Context, queues []*dao.Par
 				"created_at":               time.Now().Unix(),
 			})
 		if err != nil {
-			return fmt.Errorf("could not insert/update queue into DB: %v", err)
+			return fmt.Errorf("could not insert/update queue %s into DB: %v", q.QueueName, err)
+		}
+
+		if len(q.Children) > 0 {
+			children := []*dao.PartitionQueueDAOInfo{}
+			for _, child := range q.Children {
+				children = append(children, &child)
+			}
+			err = s.UpsertQueues(ctx, parentId, children)
+			if err != nil {
```

Comment:
I might be mistaken, but shouldn't the recursion be called with the current `queueId` as the parentID for the children? Otherwise, it seems like the `rootQueue` will be set as the parent for all generations of the queues.

Reply:
You are exactly right! This is a bad typo. Fixed in commit 41f325a.